src/http.h

```diff
@@ -35,10 +35,10 @@ class HttpStateData : public Client
         /// stores the corresponding decision
         Answers make(const Answers ans, const char *why);
 
-        Answers answer; ///< the decision id
-        const char *reason; ///< the decision reason
-        const StoreEntry *entry; ///< entry for debugging
-        const Http::StatusCode statusCode; ///< HTTP status for debugging
+        Answers answer = reuseNot; ///< the decision id
+        const char *reason = nullptr; ///< the decision reason
+        const StoreEntry *entry = nullptr;
+        const Http::StatusCode statusCode = Http::StatusCode::scNone;
```

Comment:
This _looks_ rather odd: We are declaring a constant and set it to scNone. That is not what is actually going on, but that is how this code _looks_ because the statusCode data member is (correctly) declared constant, but its default in-class initialization to scNone is ignored by the compiler in favor of the constructor code that actually sets this field to the right value (which remains constant).

The same could have been said about `entry` but that field was not declared constant.

```suggestion
        const StoreEntry * const entry;
        const Http::StatusCode statusCode;
```

Reply:
Not sure why you think this looks odd. Perhapse you have forgotten that C/C++ const'ness always "looks odd" due to types reading right-to-left.

Reply:
> Not sure why you think this looks odd

It looks odd because, normally, `const foo = scNone` means that `foo` is `scNone`; many readers would find it odd that a class needs a constant field like that -- the code could just use `scNone` directly... Here, the constant `foo` member actually gets a _different_ value elsewhere, so there is no problem other than an odd look of things.

> Perhapse you have forgotten that C/C++ const'ness always "looks odd" due to types reading right-to-left.

I have not, and this particular case reads left-to-right just fine. The problem is elsewhere (as I have detailed).